static/src/javascripts/lib/detect.js

```diff
@@ -265,8 +265,9 @@ const initPageVisibility = (): void => {
         if (evt.type in evtMap) {
             pageVisibility = evtMap[evt.type];
         } else {
-            pageVisibility = window && window.hidden ? 'hidden' : 'visible';
+            pageVisibility = document && document.hidden ? 'hidden' : 'visible';
         }
+        mediator.emit(`modules:detect:pagevisibility:${pageVisibility}`);
```

Comment:
what does that do?

Reply:
It emits an event which is picked up in `https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/ui/autoupdate.js#L212`, to immediately refresh the live blog when a page transitions from invisible->visible.

Reply:
+1